spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/TaskServiceUtils.java

```diff
@@ -128,8 +128,13 @@ public static TaskDefinition updateTaskProperties(TaskDefinition taskDefinition,
 			}
 			builder.setProperty("spring.datasource.username", dataSourceProperties.getUsername());
 		}
-		builder.setProperty("spring.datasource.url", dataSourceProperties.getUrl());
-		builder.setProperty("spring.datasource.driverClassName", dataSourceProperties.getDriverClassName());
+		if(!taskDefinition.getProperties().containsKey("spring.datasource.url")) {
+			builder.setProperty("spring.datasource.url", dataSourceProperties.getUrl());
+		}
+		if(!(taskDefinition.getProperties().containsKey("spring.datasource.driverClassName") ||
```

Comment:
Do you think we need to resolve this by Relaxed names as there are other combinations of property names users could come up with when specifying the task definition properties?

Reply:
👍